### ISSUE TITLE
Sigh tries to load gulp plugins which is not defined in sigh file

### ISSUE THREAD

**Issue by unlight** [sighjs/sigh#53]
I get error when running sigh, from `gulp-tslint`. I have `gulp-tslint` in `devDependencies`, but I do not have `tslint` var in `sigh.js`. Why sigh tries to load it, if it will not be used.
```js
var merge, glob, concat, write, env, debounce, pipeline;
var ts;
var ava;
var thru;

module.exports = function (pipelines) {
// ...
};
```

**Comment by ohjames** [comment]
ATM it tries to inject every gulp plugin, maybe I can use rewire to check which are defined.

**Comment by ohjames** [comment]
Yeah this can be fixed pretty easily. I'm busy for a week or so, I can point you in the direction of the fix if you feel like opening a PR, otherwise I'll get to it in a week or two.
